api/src/auth/drivers/ldap.ts

```diff
@@ -246,35 +227,52 @@ export class LDAPAuthDriver extends AuthDriver {
 
 		await this.validateBindClient();
 
-		const userDn = await this.fetchUserDn(payload.identifier);
+		const { userDn, userScope, userAttribute, groupDn, groupScope, groupAttribute } = this.config;
 
-		if (!userDn) {
+		const userInfo = await this.fetchUserInfo(
+			userDn,
+			new EqualityFilter({
+				attribute: userAttribute ?? 'cn',
+				value: payload.identifier,
+			}),
+			userScope ?? 'one'
+		);
+
+		if (!userInfo?.dn) {
 			throw new InvalidCredentialsException();
 		}
 
-		const userId = await this.fetchUserId(userDn);
-		const userGroups = await this.fetchUserGroups(userDn);
-
 		let userRole;
 
-		if (userGroups.length) {
-			userRole = await this.knex
-				.select('id')
-				.from('directus_roles')
-				.whereRaw(`LOWER(??) IN (${userGroups.map(() => '?')})`, [
-					'name',
-					...userGroups.map((group) => group.toLowerCase()),
-				])
-				.first();
+		if (groupDn) {
+			const userGroups = await this.fetchUserGroups(
+				groupDn,
+				new EqualityFilter({
+					attribute: groupAttribute ?? 'member',
+					value: groupAttribute?.toLowerCase() === 'memberuid' && userInfo.uid ? userInfo.uid : userInfo.dn,
```

Comment:
Is it even possible that userInfo.uid is empty after you fetched all data from ldap?

Reply:
UID can be empty on a non posix user. I thought in that instance it's better to fallback to their DN rather than fetch groups where "memberUid = undefined" which could potentially pull in any non-posix group.

Reply:
Sounds reasonable and looks good to me then